### ISSUE TITLE
Additional language page name doesn't validate if it already exists

### ISSUE THREAD

**Issue by PWaddict** [processwire/processwire-issues#322]
### Short description of the issue

If I edit the page name of an additional language it doesn't validate if it already exists.

### Expected behavior

Page name for additional language should be checked if already exists and name it properly just like it happens on the default language page name: Session: Changed page URL name to "test-1" because requested name was already taken.

### Actual behavior

Page name for an additional language saved without renaming with the "-1" if it already exists which leads to multiple pages with the same additional language page name.

### Steps to reproduce the issue

1. Create 2 pages with the same title
2. Edit the additional language's page name for the 2nd page with the same additional language's page name from the 1st page.

### Setup/Environment

* ProcessWire version: 3.0.67

**Comment by matjazpotocnik** [comment]
I'm using 3.0.80 but when I try to set the name on additional language I get:

`Session: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'one-1' for key 'name1231_parent_id'`

**Comment by PWaddict** [comment]
@matjazpotocnik I've updated the steps to reproduce the issue. Try again.
@ryancramerdesign I hope you didn't forget about this critical issue.

**Comment by PWaddict** [comment]
@ryancramerdesign The issue still exists on 3.0.94.

**Comment by matjazpotocnik** [comment]
@PWaddict I still get SQL error, but not always! Eventually, I managed to create three pages with the same name on alternate language.

**Comment by rolandtoth** [comment]
I could reproduce the issue. At the end of step 2 I got two pages where the additional language names were identical. There was no error messages when I changed page2 name to something else and then renamed back to the one that already existed.

Trying to view this second page will load the first page (of course). Using latest PW 3.095.

**Comment by ryancramerdesign** [comment]
So far I can't duplicate it here. I tried to create two pages with the same parent, where I gave each the non-default language (fr) of "bienvenues". I get the expected error notification with "SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'bienvenues-1001' for key 'name1012_parent_id'". 

Note however that it's not going to prevent you from having two pages with the same name in different languages. For instance, you might have a page in FR with name "test", and another page in DE with name "test", and that is allowed, because the actual URL will end up being /fr/test/ or /de/test/, and PW should still be able to tell what language is intended.

**Comment by rolandtoth** [comment]
Are you sure you've followed the steps in the issue description? The trick is to change the auto-generated name back to one that already exists.

[Quick screencap](http://rolandtoth.hu/pic/pw/pw-multilang-names.gif)

**Comment by matjazpotocnik** [comment]
> I get the expected error notification with "SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'bienvenues-1001' for key 'name1012_parent_id'".

If an error is expected, wouldn't it be better to prevent it in the first place or at least inform the user in more "human" wording?

**Comment by ryancramerdesign** [comment]
Thanks that screencap clarified it and I was able to duplicate it. The issue is that the page with default language name of "one" has an the alternate language name of NULL, which means "inherit from default". That's because the name was never set to something different than the default. So if you ignore the error in step 2 and modify that other page's alternate language page name to be "one", then it is allowed by the database, since NULL and "one" don't produce a key conflict. While I can't completely prevent it from occurring, I was able able to develop code that produces another error message to notify the editor more specifically of the potential conflict. It's not particularly simple to detect and involved quite a bit of code that will need more testing, so I will add this to the dev branch a little later since I'm currently prepping a master version. Thanks.

**Comment by rolandtoth** [comment]
Great, thanks for the fix!

**Comment by ryancramerdesign** [comment]
I've gone ahead and committed it to the dev branch, but the call to it is commented out. If you want to try it out, just uncomment that one added line with the `@todo` on it. I will also uncomment it on the dev branch after the merge to master.

**Comment by PWaddict** [comment]
@ryancramerdesign I've tested it on existed pages and it works great. No more duplicate page names there BUT if you already created a page with a page name /test/ and /fr/test and try to create a new page and the user type Test on both default and additional language Titles then that page's names will be /test-1/ and /fr/test which leads to duplicate page name on additional language.

**Comment by matjazpotocnik** [comment]
I tested here with an alternate language that is different than none/NULL (Slovenian as primary and English "en" as secondary). An error is thrown: `A sibling page (id=1262) is already using name "ml1" for language: English` and also `Session: A sibling page (id=1262) is already using name "ml1" for language: English (_pw_page_name)`. I think this is fixed (ok, just one error would be enough).

**Comment by matjazpotocnik** [comment]
Yes, I can confirm what PWaddict wrote. As you check for a duplicate on the default language (`Name is already taken`, `Session: Warning, the name you selected "ml1" was already in use and has been changed to "ml1-2"`) you should also check for other languages.

**Comment by ryancramerdesign** [comment]
Thanks @PWaddict and @matjazpotocnik I have pushed another commit for this which should fix that issue. Like before, this feature is not enabled by default. But unlike before, you don't have to uncomment the line yourself. instead set $config->debug=true; and $config->advanced=true; in your /site/config.php file and it'll be enabled. Once we've done more testing I'll update it to make it enabled without those conditions. Please keep this issue report open until we've done that, otherwise I might forget. :)

**Comment by PWaddict** [comment]
@ryancramerdesign There are no more duplicates. Thank you for fixing it.

**Comment by PWaddict** [comment]
@ryancramerdesign do you have an ETA when the conditions $config->debug=true; and $config->advanced=true; will be removed from your fix so we can use it on production sites?

**Comment by ryancramerdesign** [comment]
I've not had any reports of issues with this additional checking, so seems like a good time to go ahead and apply it without debug/advanced modules. I've gone ahead and pushed that update. Please let me know if you run into any troubles with it. Thanks.
